### PR TITLE
Retain calayer after creation

### PR DESCRIPTION
**Pull request by xiaopengli89**
Run `create_calayer` test:
```
signal: 11, SIGSEGV: invalid memory reference
```

### CODE REVIEW

**Comment by jdm** [comment]
@xiaopengli89 What is `create_calayer` test? I'd like to understand the code that is triggering this crash, since I worry that we might end up leaking layer objects with this change.

**Comment by xiaopengli89** [comment]
> @xiaopengli89 What is `create_calayer` test? I'd like to understand the code that is triggering this crash, since I worry that we might end up leaking layer objects with this change.

Creating `CALayer` in `autoreleasepool`, event loop or child thread will trigger this crash, in which case the runtime will hold ownership of the `CALayer`.

**Comment by nicoburns** [comment]
This fixes #643

**Review by madsmtm — APPROVED** [review]

File: cocoa/src/quartzcore.rs [hunk 1]
```diff
@@ -1674,3 +1678,13 @@ pub type CVSMPTETimeFlags = u32;
...
 pub const kCVSMPTETimeValid: CVSMPTETimeFlags = 1 << 0;
 pub const kCVSMPTETimeRunning: CVSMPTETimeFlags = 1 << 1;
+
+#[cfg(test)]
+mod test {
+    use super::CALayer;
+
+    #[test]
+    fn create_calayer() {
+        let _ = CALayer::new();
```
[the remark this answers is not in the document]
Comment (xiaopengli89) [reply cocoa/src/quartzcore.rs]: Done.

**Review by jdm — APPROVED** [review]

**Pull request merged**